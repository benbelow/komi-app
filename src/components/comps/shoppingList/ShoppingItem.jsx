// @flow
import * as React from 'react'
import styled from 'styled-components'

import palette from '../../../style/palette'

const ItemWrapper = styled.div`
  width: 100%;
  padding: 10px;
  border-bottom: 1px solid ${palette.divider};
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: ${({ isDone }) => (isDone ? palette.disabled : palette.text)};
  font-style: ${({ isDone }) => (isDone ? 'italic' : 'normal')};
  text-decoration: ${({ isDone }) => (isDone ? `line-through ${palette.disabled}` : 'none')};
  cursor: pointer;
  // transition: all 200 ms ease;
`

type Props = {
  ingredient: {
    name: string,
    quantity: number,
    nature: string
  }
}

type State = {
  isDone: boolean
}

export default class ShoppingListItem extends React.Component<Props, State> {
  state = {
    isDone: false,
  }
  render() {
    const { name, quantity, nature } = this.props.ingredient
    return (
      <ItemWrapper isDone={this.state.isDone} onClick={() => this.setState({ isDone: !this.state.isDone })}>
        <p>{name}</p>

        <p>
          {quantity} {nature}
        </p>
      </ItemWrapper>
    )
  }
}
