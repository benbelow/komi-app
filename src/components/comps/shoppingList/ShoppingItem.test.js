import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import ShoppingItem from './ShoppingItem'

describe('ShoppingItem', () => {
  const ingredient = {
    name: 'tomate',
    quantity: 2,
    nature: 'item',
  }
  it('should match snapshot', () => {
    const comp = renderer.create(<ShoppingItem ingredient={ingredient} />)
    const tree = comp.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
