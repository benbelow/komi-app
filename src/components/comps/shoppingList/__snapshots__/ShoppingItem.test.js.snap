// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingItem should match snapshot 1`] = `
.c0 {
  width: 100%;
  padding: 15px;
  border-bottom: 1px solid rgba(0,0,0,.12);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(0,0,0,.87);
  font-style: normal;
  text-decoration: none;
  cursor: pointer;
}

<div
  className="c0"
  onClick={[Function]}
>
  <p>
    tomate
  </p>
  <p>
    2
     
    item
  </p>
</div>
`;
